repos/system_upgrade/el7toel8/actors/firewalldfactsactor/actor.py

```diff
@@ -0,0 +1,56 @@
+from leapp.actors import Actor
+from leapp.models import FirewalldFacts
+from leapp.tags import FactsPhaseTag, IPUWorkflowTag
+
+from leapp.libraries.actor import private
+
+import os
+import xml.etree.ElementTree as ElementTree
+
+
+class FirewalldFactsActor(Actor):
+    """
+    Provides data about firewalld
+
+    After collecting data, a message with relevant data will be produced.
+    """
+
+    name = 'firewalld_facts_actor'
+    consumes = ()
+    produces = (FirewalldFacts,)
+    tags = (FactsPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        facts = FirewalldFacts()
+
+        try:
+            tree = ElementTree.parse('/etc/firewalld/lockdown-whitelist.xml')
+            root = tree.getroot()
+            facts.firewall_config_command = private.getLockdownFirewallConfigCommand(root)
+        except IOError:
+            pass
+
+        try:
+            tree = ElementTree.parse('/etc/firewalld/direct.xml')
+            root = tree.getroot()
+            facts.ebtablesTablesInUse = private.getEbtablesTablesInUse(root)
+        except IOError:
+            pass
```

Comment:
Just question, is it really safe to skip these exceptions (without any log)?

Reply:
Yes. The existence of those XML files is optional. In the "direct.xml" case it's even the norm.
If you want we can add a debug level log.. if that makes sense.